torch/_inductor/overrides.py

```diff
@@ -967,6 +973,54 @@ def permute_matmul_fusion(module: torch.fx.GraphModule) -> torch.fx.GraphModule:
     module.recompile()
     return module
 
+def triton_int_mm(A: torch.Tensor, B: torch.Tensor, output_dtype) -> torch.Tensor:
+    from torch._inductor.triton_ops import matmul
+    from torch._inductor.utils import has_triton
+    if has_triton() and A.dtype == torch.int8 and B.dtype == torch.int8 and output_dtype == torch.int32:
+        out = A.new_empty(A.size(0), B.size(1), dtype=output_dtype)
+        torch._triton_mm(A, B, out)
+    else:
+        out = torch.mm(A.int(), b.int())
+    return out
+
+def type_cast_matmul_fusion(module: torch.fx.GraphModule) -> torch.fx.GraphModule:
+    for node in module.graph.nodes:
+        if node.op == "call_function" and (
+            node.target == torch.bmm or node.target == torch.matmul or node.target == torch.mm
+        ):
+            normalized = NormalizedMatmulNode(node)
+            A = normalized.get_input()
+            B = normalized.get_other()
+            Atrans = Btrans = False
+            if A.op == "call_method" and A.target == "int":
```

Comment:
You should check `A.args[0]` dtype here too, if someone is converting `float` tensors to `int` you can't replace `A` with `A.args[0]`.
Also, what about `.to(torch.int)`?

Reply:
If AOT Autograd were to unconditionally run above inductor, then I think you'll also end up seeing `at::_to_copy(...)` here

Reply:
This pass is run before AOT autograd, but that's a good point, maybe a better place for this is after, when all the method type conversion calls are normalized to `_to_copy`.